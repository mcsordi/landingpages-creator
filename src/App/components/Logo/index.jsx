import P from 'prop-types';
function Logo({ image = false, logo }) {
  return (
    <div className="flex items-center">
      {image ? (
        <img src={logo} alt="logo image" className="w-28" />
      ) : (
        <span className="text-intermedium font-serrat font-900">
          {logo.toUpperCase()}
        </span>
      )}
    </div>
  );
}
Logo.propTypes = {
  logo: P.string.isRequired,
  image: P.bool,
};
export default Logo;
