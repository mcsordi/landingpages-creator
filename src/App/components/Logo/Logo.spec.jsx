import { render, screen } from '@testing-library/react';
import Logo from '.';

describe('<Logo/>', () => {
  it('should render logo text component', () => {
    render(<Logo logo="logo" />);
    const logo = screen.getByText('logo'.toUpperCase());
    expect(logo).toHaveClass('text-intermedium font-serrat font-900');
  });
  it('should render logo image component', () => {
    render(<Logo image logo="./images/landing_page_2_c6baa21725.svg" />);
    const logo = screen.getByRole('img');
    expect(logo).toHaveAttribute(
      'src',
      './images/landing_page_2_c6baa21725.svg',
    );
  });
  it('should match a logo component snapshot', () => {
    render(<Logo logo="logo" />);
    const logo = screen.getByText('logo'.toUpperCase());
    expect(logo).toMatchSnapshot();
  });
});
