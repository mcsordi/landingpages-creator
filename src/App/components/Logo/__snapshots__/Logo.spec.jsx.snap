// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Logo/> should match a logo component snapshot 1`] = `
<span
  class="text-intermedium font-serrat font-900"
>
  LOGO
</span>
`;
